core/src/main/scala/kafka/zk/KafkaZkClient.scala

```diff
@@ -402,6 +404,25 @@ class KafkaZkClient private (zooKeeperClient: ZooKeeperClient, isSecure: Boolean
     }
   }
 
+  /**
+    * Gets all brokers with broker epoch in the cluster.
+    * @return sequence of brokers in the cluster.
+    */
+  def getAllBrokerAndEpochsInCluster: Seq[(Broker, Long)] = {
+    val brokerIds = getSortedBrokerList
+    val getDataRequests = brokerIds.map(brokerId => GetDataRequest(BrokerIdZNode.path(brokerId), ctx = Some(brokerId)))
+    val getDataResponses = retryRequestsUntilConnected(getDataRequests)
+    getDataResponses.flatMap { getDataResponse =>
+      val brokerId = getDataResponse.ctx.get.asInstanceOf[Int]
+      getDataResponse.resultCode match {
+        case Code.OK =>
+          Option((BrokerIdZNode.decode(brokerId, getDataResponse.data).broker, getDataResponse.stat.getCzxid))
```

Comment:
Do we expect `BrokerIdZNode.decode(...).broker` to return null? If not, it may be simpler to just do `Some(BrokerIdZNode.decode(brokerId, getDataResponse.data).broker, getDataResponse.stat.getCzxid)`.

Reply:
Good catch. Fixed.